app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java

```diff
@@ -647,6 +647,8 @@ private void updateSearchUI() {
         outdoorDirections.deleteDirection();
         if (location != null && destination != null) {
             outdoorDirections.requestDirections();
+            GlobalPathFinder globalPathFinder = new GlobalPathFinder((MainActivity)getActivity(),location,destination);
+            globalPathFinder.run();
```

Comment:
nope. this has to be run in a seperate thread. run() runs it in the current one.